tests/index_routines.py

```diff
@@ -24,6 +24,438 @@
 from legate.core import LEGATE_MAX_DIM
 
 
+def advanced_indexing():
```

Comment:
Can you add tests for some more cases? I believe these are not covered currently, but it's possible I missed something:

* get_item: transformed index array
* set_item: transformed base array
* set_item: transformed index array
* set_item: transformed rhs array
* two boolean arrays of the same shape
  ```
  >>> y = np.ones((3,4,3,4))
  >>> z = np.array([[True,False,False,False], [False,False,False,False], [False,False,False,True]])
  >>> y[z,z].shape
  (2,)
  ```
* two boolean arrays that must be broadcasted together
  ```
  >>> y = np.ones((3,4,3,4))
  >>> z = np.array([[True,False,False,False], [False,False,False,False], [False,False,False,True]])
  >>> y[z, [True,False,False]].shape
  (2, 4)
  ```
* boolean array not at index 0
  ```
  >>> x = np.ones((2,3,4,5))
  >>> b0 = np.full((3,4), True)
  >>> x[:,b0].shape
  (2, 12, 5)
  ```
* two boolean arrays separated by `:`
  ```
  >>> x = np.ones((2,3,4,5))
  >>> x[:, [True,True,True], :, [True,True,True,False,False]].shape
  (3, 2, 4)
  ```
* two boolean arrays separated by scalar
  ```
  >>> x = np.ones((2,3,4,5))
  >>> x[:, [True,True,True], 1, [True,True,True,False,False]].shape
  (2, 3)
  ```
* base array of a different datatype (e.g. `np.int16`)
* index array of a different datatype (e.g. `np.int16`)
* index array of a different datatype (e.g. `np.int16`)
* mixing bool and int index arrays
* an example with `np.newaxis`